### PR TITLE
Feat/utils

### PR DESCRIPTION
**Pull request by Debarshi95**
Added Utility function, CRUD operations and related components

Preview:https://627830d75dcde40e6d6bef34--debarshib-chatter.netlify.app/

### CODE REVIEW

File: src/components/CardHeader/CardHeader.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { Text, Button } from 'components';
+import { getFirstChar } from 'utils/helperFuncs';
+import { requestUpdateUserProfileData } from 'store/reducers/slices';
+
+const CardHeader = ({ authUserId, username, userId, showFollowButton, isFollowing }) => {
+  const dispatch = useDispatch();
+
+  const handleOnClick = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
```
Comment (saurabhchirde) [inline src/components/CardHeader/CardHeader.js]: this is something new to me.

**Review by saurabhchirde — APPROVED** [review]
Couldn't find any issue, I appreciate your code structure and the written code practice, get to learn many new things from your PRs. 
You can proceed to merge.

File: src/store/reducers/slices/postSlice.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+export const requestCreatePost = createAsyncThunk(
+  'post/createPost',
+  async (postData, { rejectWithValue, dispatch }) => {
+    const { userId, content, following, username } = postData;
+    try {
+      const res = await createPost({ userId, content, username });
+      if (res?.id) {
+        await Promise.all([
+          dispatch(requestGetAllPosts({ userId, following })),
```
Comment (sanjitsarkar) [inline src/store/reducers/slices/postSlice.js]: **suggestion:** there's no need to fetch all the posts after creating post just unshift or insert the post data into the state in the extraReducers like this 
```jsx
addCase(createPost.fulfilled, (state, action) => {
        state.loading = false;
        state.posts.unshift(action.payload.post);

      })
```
and in this function just return the data as post object

File: src/utils/helperFuncs.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const getFirstChar = (str = '') => str.charAt(0).toUpperCase();
+
+export const isFollowing = (user, authUserId) => {
+  return user?.followers?.find((followers) => followers === authUserId) || false;
```
Comment (sanjitsarkar) [inline src/utils/helperFuncs.js]: **Suggestion:** use Array method some instead of find to only return true if found otherwise false

**Review by sanjitsarkar — APPROVED** [review]

File: src/store/reducers/slices/postSlice.js (hunk 2 above)
Comment (Debarshi95) [reply src/store/reducers/slices/postSlice.js:33]: Thanks for the comment but pushing directly into the state before persisting is not recommended. What if the post creation in the database was not successful and the user refreshes the page? They won't see the latest post they created.

**Pull request merged**